Utils/github_workflow_scripts/autobump_rn.py

```diff
@@ -0,0 +1,1170 @@
+    def check_update_type(
+        prev_version: Version, new_version: Version
+    ) -> Optional[UpdateType]:
+        """Checks what was the update type when the release notes were generated.
+        Args:
+            prev_version: the pack version before updating release notes.
+            new_version: the pack version after updating release notes.
+        Returns:
+            The pack update type if the update type was legal.
+        """
+        same_major = prev_version.major == prev_version.major
+        same_minor = prev_version.minor == prev_version.minor
+        if prev_version.micro + 1 == new_version.micro and same_minor and same_major:
+            return UpdateType.REVISION
+        elif (
+            prev_version.minor + 1 == new_version.minor
+            and same_major
+            and not new_version.micro
+        ):
+            return UpdateType.MINOR
+        elif (
+            prev_version.major + 1 == new_version.major
+            and not new_version.minor
+            and not new_version.micro
+        ):
+            return UpdateType.MAJOR
+        else:
+            return None
+
+
+class PackAutoBumper:
+    def __init__(
+        self,
+        pack_id: str,
+        rn_file_path: Path,
+        update_type: UpdateType,
+    ):
+        """
+        Autobump pack version.
+        Args:
+            pack_id: Pack id to its release notes.
+            rn_file_path: last release notes path.
+            update_type: the update type that was in the pr.
+        """
+        self.pack_id = pack_id
+        self._last_rn_file_path = rn_file_path
+        self._update_type = update_type
+        self._update_rn_obj = UpdateRN(
+            pack_path=f"{PACKS_DIR}/{pack_id}",
+            update_type=update_type.value,
+            modified_files_in_pack=set(),
+            added_files=set(),
+            pack=pack_id,
+            is_force=True,
+        )
+        # Setting to default. Will be updated once we are checked out to the branch in set_pr_changed_rn_related_data.
+        self._bc_file = Path(str(self._last_rn_file_path).replace("md", "json"))
+        self._has_bc = False
+        self._rn_text = ""
+        self._bc_text = ""
+
+    def set_pr_changed_rn_related_data(self):
+        """Opens release notes and bc changes files and saves its text."""
+        self._rn_text = self._last_rn_file_path.read_text()
+        self._has_bc = self._bc_file.is_file()
+        if self._has_bc:
+            self._bc_text = self._bc_file.read_text()
+
+    def autobump(self) -> str:
+        """AutoBumps packs version:
+        1. Bumps numeric version of the pack
+        2. Writes new version to metadata
+        3. Creates new release notes file
+        4. Writes previous release notes content to new path
+        5. If there breaking changes file, updating it to the new version
+        Returns: (str) new pack version.
+        """
+
+        print(f"Starting to bump packs {self.pack_id} version.")
```

Comment:
please replace prints with `logging`

Reply:
Why `logging` is better? We don't use logging in no one of our GitHub actions. Logging handling parallel printing. this script is sequential, no parallel printings here.